editor/liveblocks.config.ts

```diff
@@ -98,14 +98,31 @@ export type RoomEvent = ControlChangedRoomEvent
 export type ThreadMetadata = {
   // quote: string;
   // time: number;
-  type: 'canvas'
-  x: number // x and y is global when sceneId is undefined, and local to the scene when sceneId is not null
+  x: number
   y: number
   sceneId?: string
+  sceneX?: number
```

Comment:
I don't like this! what's the point of these being optional here and then have a separate SceneThreadMetadata where they are declared as non optional?

Reply:
The reason is that canvas comments don't have these properties (sceneId, sceneX, sceneY), and we are only allowed to have string, number and boolean properties here, so no way to create a structured type.

Reply:
`why not `CanvasThreadMetadata` without these fields and `SceneThreadMetadata` with these fields, and then `ThreadMetadata = CanvasThreadMetadata | SceneThreadMetadata`

but better yet, why not make it a discriminated union type with a type: 'canvas' | 'scene'? I see on master there was some notion to do that

Reply:
I can not use discriminated union types because I can only add string/boolean/number fields.
Undiscrimated union types are not that nice to work with, but you can be right maybe this solution is better: 68b241864f82462319480bf0d845bfdef0f00511

Reply:
It turned out the problem was not discriminated union types, but that I used interfaces and not types.... Here is the proper solution: dcc67bdbb4071c8c5dd7c6c7e6b4db5f94202cf2